tests/sentry/testutils/pytest/mocking/test_mocking.py

```diff
@@ -47,3 +49,20 @@ def test_return_values_as_dict(self):
             a_function_that_calls_get_cat()
             assert get_cat_spy.call_count == 1
             assert return_values["get_cat"][0] == "piper"
+
+    def test_records_thrown_exception(self):
+        erroring_get_dog_results: list[Any] = []
+
+        wrapped_erroring_get_dog = capture_results(erroring_get_dog, erroring_get_dog_results)
+
+        with mock.patch(
+            "tests.sentry.testutils.pytest.mocking.animals.erroring_get_dog",
+            wraps=wrapped_erroring_get_dog,
+        ) as erroring_get_dog_spy:
+            a_function_that_calls_erroring_get_dog()
+
+            assert erroring_get_dog_spy.call_count == 1
+
+            # Need to use `repr` since even identical errors don't count as equal
+            result = erroring_get_dog_results[0]
+            assert repr(result) == "TypeError('Expected dog, but got cat instead.')"
```

Comment:
Nit: Alternative here is
`type(e) is type(e2) and e.args == e2.args`

Reply:
Yeah, you're right, that's nicer. And it's actually what I'm doing in my other tests, so IDK why I didn't think of it here. Anyway, fixed.